src/try_with_timeout.jl

```diff
@@ -34,7 +34,31 @@ it. If `f` completes successfully, return its result.
 This allows the calling function to check whether the timeout has been reached
 by checking `x[]` and if `true`, the timeout was reached and the function can
 cancel/abort gracefully. The 3rd argument `T` is optional (default `Any`) and
-allows passing an expected return type that `f` should return.
+allows passing an expected return type that `f` should return; this allows avoiding
+a dynamic dispatch from non-inferability of using `try_with_timeout` with `f`.
+
+# Examples
+
+```julia
+julia> try_with_timeout(_ -> 1, 1)
+1
+
+julia> try_with_timeout(_ -> sleep(3), 1)
+ERROR: TimeoutException: try_with_timeout timed out after 1.0 seconds
+Stacktrace:
+ [1] try_with_timeout(::var"#1#2", ::Int64) at ./REPL[1]:1
+ [2] top-level scope at REPL[2]:1
+
+julia> try_with_timeout(_ -> error("hey"), 1)
+ERROR: hey
+Stacktrace:
+ [1] error(::String) at ./error.jl:33
+ [2] (::var"#1#2")(::TimedOut{Any}) at ./REPL[1]:1
+ [3] try_with_timeout(::var"#1#2", ::Int64) at ./REPL[1]:1
+ [4] top-level scope at REPL[3]:1
+
+julia> try_with_timeout(_ -> 1, 1, Int)
+1
```

Comment:
need the closing triple back-ticks here (though github doesn't let me suggest it)

Reply:
good catch